internal/controllers/machine/machine_controller_noderef.go

```diff
@@ -87,6 +87,7 @@ func (r *Reconciler) reconcileNode(ctx context.Context, cluster *clusterv1.Clust
 		}
 		log.Error(err, "Failed to retrieve Node by ProviderID")
 		r.recorder.Event(machine, corev1.EventTypeWarning, "Failed to retrieve Node by ProviderID", err.Error())
+		conditions.MarkUnknown(machine, clusterv1.MachineNodeHealthyCondition, clusterv1.NodeInspectionFailedReason, "Failed to get the node which is hosting this component: %v", err)
```

Comment:
```suggestion
		conditions.MarkUnknown(machine, clusterv1.MachineNodeHealthyCondition, clusterv1.NodeInspectionFailedReason, "Failed to get the node")
```

Sorry just realized now:
1. what does "hosting this component" mean? (which component?)
2. we should probably not include the error in the condition, we already log it. The problem with adding the error message to the condition is that if the error is slightly different on reconcile we end up with an "infinite reconcile" (i.e. continuous reconciles of the Machine object as the condition always changes)

Reply:
> we should probably not include the error in the condition, we already log it. The problem with adding the error message to the condition is that if the error is slightly different on each reconcile we end up with an "infinite reconcile" (i.e. continuous reconciles of the Machine object as the condition always changes)

Understood. Then should bellow code be fixed as you mentioned, too?

https://github.com/kubernetes-sigs/cluster-api/blob/e24d2731669fc7003cfa6ed1a034530bca34c882/controlplane/kubeadm/internal/workload_cluster_conditions.go#L261

Reply:
> It comes from enxebre's https://github.com/kubernetes-sigs/cluster-api/pull/8268#discussion_r1133772992

I see. I think in the other case "hosting this component" makes sense, but here it doesn't.


> Understood. Then should bellow code be fixed as you mentioned, too?

I would say yes. (also the occurence right above this one in l.258)

Would be good if we can add an additional log in ~ l.255 so that the error shows up in the logs. Just like we do here in l.88